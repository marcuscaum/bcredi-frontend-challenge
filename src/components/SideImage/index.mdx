---
name: SideImage
---


import { Playground, PropsTable } from 'docz'
import SideImage from './index'

# SideImage

<PropsTable of={SideImage} />

## Basic usage

<Playground>
  <SideImage
    messageAuthor={
      <React.Fragment>
        <strong>Camila Bragança,</strong>
        <br />
        Sideral Tecnologia
      </React.Fragment>
    }
    message="Obtive crédito para capital de giro. O processo foi bem sucedido e tudo que foi abordado, foi cumprido."
  />
</Playground>