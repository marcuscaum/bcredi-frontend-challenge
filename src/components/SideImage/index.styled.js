import styled from 'styled-components';
import SignUpImage from 'assets/signup_page_image.png';

export const MessageContainer = styled.div`
  display: flex;
  flex-flow: row-reverse;
  position: absolute;
  width: 100%;
  height: 100%;
  background-image: linear-gradient(
    to bottom,
    rgba(0, 0, 0, 0.2),
    rgba(0, 0, 0, 0) 17%,
    rgba(0, 0, 0, 0) 44%,
    rgba(0, 0, 0, 0.5) 72%,
    rgba(0, 0, 0, 0.8)
  );
`;

export const MessageBox = styled.div`
  display: flex;
  align-self: flex-end;
  flex-direction: column;
  margin: 0 40px 40px 0;
  width: 370px;
  color: white;
  text-align: right;
  line-height: 1.5;
`;

export const MessageContent = styled.p`
  font-size: 20px;
  margin: 0;
`;

export const MessageAuthor = styled.p`
  font-size: 16px;
  margin: 0;
  margin-top: 8px;
`;

export const SideImage = styled.div`
  display: flex;
  position: relative;
  height: 100vh;
  width: 550px;
  background-size: cover;
  background-position: center center;
  background-image: url('${SignUpImage}');
`;
