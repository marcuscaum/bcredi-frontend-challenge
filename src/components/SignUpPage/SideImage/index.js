import React from 'react';
import PropTypes from 'prop-types';

import {
  SideImage,
  MessageContainer,
  MessageBox,
  MessageContent,
  MessageAuthor,
} from './index.styled';

const SideImageComponent = ({ message, messageAuthor }) => (
  <SideImage>
    <MessageContainer>
      <MessageBox>
        <MessageContent>{message}</MessageContent>
        <MessageAuthor>{messageAuthor}</MessageAuthor>
      </MessageBox>
    </MessageContainer>
  </SideImage>
);

SideImageComponent.propTypes = {
  message: PropTypes.oneOfType([PropTypes.string, PropTypes.node]).isRequired,
  messageAuthor: PropTypes.oneOfType([PropTypes.string, PropTypes.node]).isRequired,
};

export default SideImageComponent;
