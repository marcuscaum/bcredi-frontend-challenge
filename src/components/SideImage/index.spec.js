import React from 'react';
import { shallow } from 'enzyme';

import SideImage from '.';

const setup = props => shallow(<SideImage {...props} />);

describe('[Component] SideImage', () => {
  it('should render the component', () => {
    const wrapper = setup({
      message: 'Test message',
      messageAuthor: 'Test message author',
    });
    expect(wrapper).toMatchSnapshot();
  });
});
