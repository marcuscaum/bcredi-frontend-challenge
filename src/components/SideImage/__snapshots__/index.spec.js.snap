// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Component] SideImage should render the component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SideImageComponent
    message="Test message"
    messageAuthor="Test message author"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <styled.div>
        <styled.div>
                <styled.p>
                        Test message
                </styled.p>
                <styled.p>
                        Test message author
                </styled.p>
        </styled.div>
</styled.div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <styled.div>
          <styled.p>
                    Test message
          </styled.p>
          <styled.p>
                    Test message author
          </styled.p>
</styled.div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <styled.p>
              Test message
</styled.p>,
            <styled.p>
              Test message author
</styled.p>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Test message",
            },
            "ref": null,
            "rendered": "Test message",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Test message author",
            },
            "ref": null,
            "rendered": "Test message author",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <styled.div>
          <styled.div>
                    <styled.p>
                              Test message
                    </styled.p>
                    <styled.p>
                              Test message author
                    </styled.p>
          </styled.div>
</styled.div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <styled.div>
            <styled.p>
                        Test message
            </styled.p>
            <styled.p>
                        Test message author
            </styled.p>
</styled.div>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <styled.p>
                Test message
</styled.p>,
              <styled.p>
                Test message author
</styled.p>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Test message",
              },
              "ref": null,
              "rendered": "Test message",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Test message author",
              },
              "ref": null,
              "rendered": "Test message author",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
